python/pyarrow/ipc.pxi

```diff
@@ -845,6 +914,19 @@ cdef class RecordBatchReader(_Weakrefable):
         return self
 
 
+cdef void pycapsule_stream_deleter(object stream_capsule):
+    cdef:
+        ArrowArrayStream* stream
+    # Do not invoke the deleter on a used/moved capsule
+    stream = <ArrowArrayStream*>cpython.PyCapsule_GetPointer(
```

Comment:
```suggestion
    stream = <ArrowArrayStream*>PyCapsule_GetPointer(
```

(since you imported those functions as such as the top of the file, although it seems the cpython. also works)